File: app/src/main/java/tech/thdev/mediaprojectionexample/base/view/BaseMediaProjectionActivity.java
package tech.thdev.mediaprojectionexample.base.view;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;
import android.support.v7.widget.Toolbar;

import butterknife.BindView;
import butterknife.ButterKnife;
import tech.thdev.media_projection_library.view.MediaProjectionAccessActivity;
import tech.thdev.mediaprojectionexample.R;

/**
 * Created by Tae-hwan on 4/25/16.
 */
public abstract class BaseMediaProjectionActivity extends MediaProjectionAccessActivity {

    @Nullable
    @BindView(R.id.toolbar)
    Toolbar toolBar;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(getContentLayoutResource());

        ButterKnife.bind(this);

        setToolBar();
        onCreate();
    }

    /**
     * ToolBar init
     */
    protected void setToolBar() {
        if (toolBar != null) {
            if (getToolbarTitle() > 0) {
                toolBar.setTitle(getToolbarTitle());
            }
            setSupportActionBar(toolBar);
        }
    }

    @LayoutRes
    protected abstract int getContentLayoutResource();

    protected abstract void onCreate();

    protected
    @StringRes
    int getToolbarTitle() {
        return 0;
    }
}

File: app/src/main/java/tech/thdev/mediaprojectionexample/base/ui/BaseMediaProjectionActivity.kt
package tech.thdev.mediaprojectionexample.base.ui

import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.annotation.StringRes
import androidx.appcompat.widget.Toolbar
import tech.thdev.media_projection_library.ui.MediaProjectionAccessActivity
import tech.thdev.mediaprojectionexample.R

/**
 * Created by Tae-hwan on 4/25/16.
 * @since 4/12/20 Edit kotlin.
 */
abstract class BaseMediaProjectionActivity(
    @LayoutRes private val layoutRes: Int,
    @StringRes private val titleRes: Int = 0
) : MediaProjectionAccessActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutRes)

        setToolbar()
    }

    private fun setToolbar() {
        if (titleRes > 0) {
            findViewById<Toolbar>(R.id.toolbar).run {
                setTitle(titleRes)
                setSupportActionBar(this)
            }
        }
    }
}